contracts/common/lib/DepositsTracker.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-FileCopyrightText: 2025 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+// solhint-disable-next-line
+pragma solidity >=0.8.9 <0.9.0;
+
+/// @notice Deposit information between two slots
+/// Pack slots information
+struct DepositedEthState {
+    /// tightly packed deposit data ordered from older to newer by slot
+    uint256[] slotsDeposits;
+    /// Index of the last read
+    uint256 cursor;
+}
+
+/// @notice Deposit in slot
+struct SlotDeposit {
+    /// Ethereum slot
+    uint64 slot;
+    /// cumulative sum up to and including this slot
+    uint192 cumulativeEth;
+}
+
+library SlotDepositPacking {
+    function pack(SlotDeposit memory deposit) internal pure returns (uint256) {
+        // return (uint256(deposit.slot) << 128) | uint256(deposit.depositedEth);
+        return (uint256(deposit.slot) << 192) | uint256(deposit.cumulativeEth);
+    }
+
+    function unpack(uint256 value) internal pure returns (SlotDeposit memory slotDeposit) {
```

Comment:
it's more optimal not use struct, as it located in memory and use 2x  uint256, but use 2 vars direct, as it stores in stack which is much cheaper in terms of gas.
e.g. `returns (uint64 slot, uint192 cumulativeEth)`